Source/WebKit/UIProcess/ProcessThrottler.cpp

```diff
@@ -159,6 +159,9 @@ std::optional<ProcessAssertionType> ProcessThrottler::assertionTypeForState(Proc
 
 void ProcessThrottler::setThrottleState(ProcessThrottleState newState)
 {
+    if (m_state == newState && (m_assertion && m_assertion->isValid()))
```

Comment:
Moving this early exit check back to the start of the function where it used to be. I believe the race started failing more often because I moved this further down in the function.